src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -5,345 +5,132 @@
-        case InferenceEngine::Precision::U64:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::U64>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedOutputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::I64:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::I64>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedOutputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::U8:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::U8>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedOutputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::BOOL:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::BOOL>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedOutputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        default:
-            IE_THROW(NotImplemented) << "Unsupported input precision " << it.second->getTensorDesc().getPrecision();
-        }
-        _outputs[it.first] = res;
+
+    for (const auto& it : get_outputs()) {
+        auto name = ov::op::util::get_ie_output_name(it.get_node_shared_ptr()->input_value(0));
+        ov::Tensor res;
+        res = create_shared_tensor_on_batched_tensor(m_batched_request_wrapper->_infer_request_batched->get_tensor(it),
+                                                     name,
+                                                     batched_outputs,
+                                                     m_batch_id,
+                                                     m_batch_size);
+        set_tensor(it, res);
     }
 }
-void SyncInferRequest::SetBlobsToAnotherRequest(InferenceEngine::SoIInferRequestInternal& req) {
-    for (const auto& it : _networkInputs) {
-        auto& name = it.first;
+
+void SyncInferRequest::set_tensors_to_another_request(std::shared_ptr<ov::IAsyncInferRequest>& req) {
+    for (const auto& it : get_inputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        auto blob = GetBlob(name);
-        if (req->GetBlob(name) != blob)
-            req->SetBlob(name, blob);
+        auto tensor = get_tensor(it);
+        auto type = tensor.get_element_type();
+        if (req->get_tensor(it).data(type) != tensor.data(type)) {
+            req->set_tensor(it, ov::Tensor(it, tensor.data()));
```

Comment:
Is it any problem if we set tensor without construct new one?

Reply:
It will cause the release failed, when app exit.